src/client/NetworkingClient.cpp

```diff
@@ -44,16 +45,16 @@ sf::Uint32 NetworkingClient::Start() {
     return mClientId;
 }
 
-std::vector<position> NetworkingClient::getPositions() {
-    while (!mAcceptingPositionRead) {
-        //
+std::vector<network_game_object> NetworkingClient::getNetworkGameObjects() {
+    if (mAcceptingNetworkGameObjectsRead) {
+        return mNetworkGameObjects;
     }
-    return mPositions;
+    return std::vector<network_game_object> {};
```

Comment:
why is this better?

Reply:
Rather than passing just positions (id, x_dir, y_dir) over the network we are now passing data that describes a game object (essentially a circle). 
In the previous code we we had a hardcoded `size` value, now this is sent by the server - so if there is a group of size 2, it'll actually look big on the client.